src/app/mytools/sidebar-accordion/sidebar-accordion.component.ts

```diff
@@ -1,24 +1,74 @@
+import { Component, Input, OnInit, OnChanges, SimpleChanges } from '@angular/core';
 import { DockstoreTool } from 'app/shared/swagger';
 import { Observable } from 'rxjs';
 import { ToolQuery } from '../../shared/tool/tool.query';
 import { OrgToolObject } from '../my-tool/my-tool.component';
+import { KeyValue } from '@angular/common';
+import { MetadataService } from '../../shared/swagger/api/metadata.service';
+
+interface GroupEntriesByRegistry {
+  groupEntryInfo: OrgToolObject<DockstoreTool>[];
+  registryTitle: string;
+}
 
 @Component({
   selector: 'app-sidebar-accordion',
   templateUrl: './sidebar-accordion.component.html',
-  styleUrls: ['./sidebar-accordion.component.scss'],
+  styleUrls: ['./sidebar-accordion.component.scss', '../../shared/styles/my-entry-sidebar.scss'],
 })
-export class SidebarAccordionComponent implements OnInit {
-  constructor(private toolQuery: ToolQuery) {}
+export class SidebarAccordionComponent implements OnInit, OnChanges {
   @Input() openOneAtATime;
   @Input() groupEntriesObject: OrgToolObject<DockstoreTool>[];
   @Input() refreshMessage;
-
   public toolId$: Observable<number>;
+  private dockerRegistryMap = [];
+  public registryToTools: Map<string, GroupEntriesByRegistry> = new Map<string, GroupEntriesByRegistry>([]);
   activeTab = 0;
 
-  ngOnInit(): void {
+  constructor(private toolQuery: ToolQuery, private metadataService: MetadataService) {
+    this.metadataService.getDockerRegistries().subscribe((map) => {
+      this.dockerRegistryMap = map;
+      this.dockerRegistryMap.forEach((registry) => {
+        // Do not create new keys for amazon and seven bridges, to be put in additional category
+        if (registry.dockerPath !== null && registry.dockerPath !== 'public.ecr.aws')
+          this.registryToTools.set(registry.dockerPath, { groupEntryInfo: [], registryTitle: registry.friendlyName.toUpperCase() });
+      });
+      // For additional tools, to be removed once amazon and seven bridges have their own groupings
+      this.registryToTools.set('additional', { groupEntryInfo: [], registryTitle: 'ADDITIONAL' });
+      this.sortByRegistries();
+    });
+  }
+
+  /**
+   * Display in original ordering when iterating through keys
+   */
+  public defaultOrdering(_left: KeyValue<any, any>, _right: KeyValue<any, any>): number {
+    return 0;
+  }
+
+  /**
+   * Sort tools by registries to display by groups
+   */
+  public sortByRegistries() {
+    this.groupEntriesObject.forEach((groupEntryObject) => {
+      if (this.registryToTools.has(groupEntryObject.registry)) {
+        this.registryToTools.get(groupEntryObject.registry).groupEntryInfo.push(groupEntryObject);
+      } else {
+        this.registryToTools.get('additional').groupEntryInfo.push(groupEntryObject);
+      }
+    });
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.groupEntriesObject && this.groupEntriesObject) {
+      for (let key of this.registryToTools.keys()) {
```

Comment:
Suggest `const` instead of `let`. FYI, it may seem trivial/nit-picky, and to a degree it is -- but when I saw `let`, I scanned the code to see where you modified it (but you don't). Whereas if I see `const`, I don't scan the code to see where it gets changed, because it's not allowed. It helps readers comprehend the code a little more easily.